src/api.d.ts

```diff
@@ -6,6 +6,15 @@
 import { AzExtParentTreeItem, AzExtTreeDataProvider, AzExtTreeItem, IActionContext, ISubscriptionContext, TreeItemIconPath } from '@microsoft/vscode-azext-utils';
 import * as vscode from 'vscode';
 
+export interface AzureResourceGroupsExtensionApi {
+    tree: AzExtTreeDataProvider;
```

Comment:
It probably doesn't make much difference in the grand scheme of things, but we could make all of these `readonly`.